### ISSUE TITLE
virtual population bug: array doesn't work as expected

### ISSUE THREAD

**Issue by TrejGun** [Automattic/mongoose#4285]
```js
var mongoose = require("mongoose");
mongoose.set("debug", true);

var db = mongoose.createConnection();
db.open("mongodb://localhost/issue-123");

var UserSchema = new mongoose.Schema({
	name: {
		type: String
	}
}, {
	timestamps: true,
	versionKey: false
});

UserSchema.virtual("group", {
	ref: "Group",
	localField: "_id",
	foreignField: "users",
	justOne: true
});

var User = db.model("User", UserSchema);

var GroupSchema = new mongoose.Schema({
	name: {
		type: String
	},
	users: [{
		type: mongoose.Schema.Types.ObjectId,
		ref: "User"
	}]
}, {
	timestamps: true,
	versionKey: false
});


var Group = db.model("Group", GroupSchema);

User.create({name: "Trej"})
	.then(function (user) {
		console.log("User", user);
		return Group.create({name: "my group", users: [user._id/* , mongoose.Types.ObjectId()*/]}) // <- uncomment me!
			.then(function (group) {
				console.log("Group", group);
				return User.findOne({_id: user._id}).populate({path: "group"})
					.then(function (result) {
						console.log("result", result);
						console.log("result.group", result.group); // <- null
					});
			});
	});
```

setup is same as in #4284 
then for external model with array with only desired _id everything works properly
but if there are two or more _id's in array population fails

**Comment by vkarpov15** [comment]
Fixed with #4284's fix 1eeddb567dccea704f64431a7e95b1535eba8b58

**Comment by TrejGun** [comment]
ahaha thx, i was about to push my fix
